lib/internal/histogram.js

```diff
@@ -53,6 +54,30 @@ function isHistogram(object) {
   return object?.[kHandle] !== undefined;
 }
 
+function* infinite() {
+  let id = 0;
+
+  while (true) {
+    yield id++;
+  }
+}
+
+const idWeakMap = new SafeWeakMap();
```

Comment:
Avoid using a SafeWeakMap, they mess a bit with garbage collection. I think you can store the id directly inside the Histogram.

Reply:
Should I add the id field in the C++ code? I'm not sure about how to deal with this

Reply:
I'm thinking to cache 'timerified' function in the `histogram` class, so we don't need to modify the original function.

If `options.histogram` not provided, we use a global `perf_hooks.perfomance[kTimerified]: Map` or something else, but eventually we need somewhere to store these stuff if we need cache

Reply:
I would attach them to the passed function using a private Symbol.